server/gitrest/packages/gitrest-base/src/utils/helpers.ts

```diff
@@ -178,5 +178,5 @@ export function logAndThrowApiError(error: any, request: Request, params: IRepoM
     // 400 by default, using something different here would override the expected behavior and cause issues. Because
     // of that, for now, we use 400 here. But ideally, we would revisit every RepoManager API and make sure that API
     // is actively throwing NetworkErrors with appropriate status codes according to what the protocols expect.
-    throw new NetworkError(error?.code ?? 400, `Error when processing ${request.method} request to ${request.url}`);
```

Comment:
Perhaps `typeof error?.code === number ? error.code : 400`? This would allow for numeric Error codes( e.g. internal `throw NetworkErrors`), and switch string codes to 400 (e.g. internal iso-git errors)

Reply:
So for the internal `NetworkError`s, there is a

```ts
if (isNetworkError(error)) {
        throw error;
    }
```

just a few lines above. For the other errors, they might not have any relevance to the caller, so throwing a default 400 might simplify things. What do you think?

Reply:
Oh great point. Didn't expand the above code! Thanks, this is great then :)